command/agent/command.go

```diff
@@ -260,6 +261,13 @@ func (c *Command) readConfig() *Config {
 		}
 	}
 
+	// Check to see if we should read the Vault namespace from the environment
+	if config.Vault.Namespace == "" {
+		if ns, ok := os.LookupEnv("VAULT_NAMESPACE"); ok {
```

Comment:
Could just use `config.Vault.Namespace = os.GetEnv("VAULT_NAMESPACE")` as it defaults to `""`.